jaxopt/_src/lbfgs.py

```diff
@@ -256,19 +256,35 @@ def init_state(self,
     Returns:
       state
     """
+    if isinstance(init_params, base.OptStep):
+      # this allows to maintain backward compatibility with the previous API
```

Comment:
```suggestion
      # `init_params` can either be a pytree or an OptStep object
```
I fear that your comment regarding "the old" API will be difficult to understand in a few years.

Also, can you update the docstring of `run` to mention that OptStep objects are supported for `init_params`?